src/System Application/App/Agent/Setup/AgentAccessControls/SelectAgentAccessCtrlPart.Page.al

```diff
@@ -0,0 +1,219 @@
+                    CurrPage.Update(false);
+                end;
+            }
+        }
+    }
+
+    trigger OnOpenPage()
+    var
+        AgentUtilities: Codeunit "Agent Utilities";
+    begin
+        AgentUtilities.BlockPageFromBeingOpenedByAgent();
+    end;
+
+    trigger OnAfterGetRecord()
+    begin
+        UpdateGlobalVariables();
+    end;
+
+    trigger OnDeleteRecord(): Boolean
+    begin
+        VerifyOwnerExistsInTempTable();
+        exit(true);
+    end;
+
+    trigger OnInsertRecord(BelowxRec: Boolean): Boolean
+    begin
+        Rec."Agent User Security ID" := AgentUserSecurityID;
+
+        // Default company name if not showing company field
+        if (Rec."Company Name" = '') and not ShowCompanyField then
+#pragma warning disable AA0139
+            Rec."Company Name" := CompanyName();
+#pragma warning restore AA0139
+
+        // Update GlobalSingleCompanyName if transitioning to multi-company
+        if (GlobalSingleCompanyName <> '') and (Rec."Company Name" <> GlobalSingleCompanyName) then
+            GlobalSingleCompanyName := '';
+
+        exit(true);
+    end;
+
+    internal procedure Initialize(NewAgentUserSecurityId: Guid; var TempAgentAccessControl: Record "Agent Access Control" temporary)
+    var
+        AgentSingleCompany: Boolean;
+    begin
+        AgentUserSecurityID := NewAgentUserSecurityId;
+        Rec.Copy(TempAgentAccessControl, true);
+
+        AgentSingleCompany := AgentImpl.TryGetAccessControlForSingleCompany(AgentUserSecurityID, GlobalSingleCompanyName);
+        if not ShowCompanyFieldOverride then
+            ShowCompanyField := not AgentSingleCompany;
+
+        CurrPage.Update(false);
+    end;
+
+    local procedure UpdateGlobalVariables()
+    var
+        User: Record "User";
+    begin
+        Clear(UserFullName);
+        Clear(UserName);
+
+        if IsNullGuid(Rec."User Security ID") then
+            exit;
+
+        if not User.Get(Rec."User Security ID") then
+            exit;
+
+        UserName := User."User Name";
+        UserFullName := User."Full Name";
+    end;
+
+    local procedure ValidateUserName(NewUserName: Text)
+    var
+        UserSecurityID: Guid;
+    begin
+        if not FindUserByName(NewUserName, UserSecurityID) then
+            exit;
+
+        Rec.Validate("User Security ID", UserSecurityID);
+        UpdateGlobalVariables();
+    end;
+
+    local procedure FindUserByName(NewUserName: Text; var UserSecurityID: Guid): Boolean
+    var
+        User: Record User;
+        UserGuid: Guid;
+    begin
+        if Evaluate(UserGuid, NewUserName) then begin
+            if not User.Get(UserGuid) then
+                exit(false);
+            UserSecurityID := User."User Security ID";
+            exit(true);
+        end;
+
+        User.SetRange("User Name", NewUserName);
+        if not User.FindFirst() then begin
+            User.SetFilter("User Name", '@*''''' + NewUserName + '''''*');
```

Comment:
Please double check this code, it looks wrong to me. 
This filter means that we will find a substring
e.g. if the user name is Sotiris in the system
if I type Iris I will get Sotiris

I would expect that autocomplete should use a prefix not any part of the string.